webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/useTableData.tsx

```diff
@@ -140,22 +207,22 @@ export function useTableData(
         return this.view.getCellValue(key);
       },
       getColumnIndexFromKey(key) {
-        return this.columns.findIndex(column => column.key === key);
+        return columns.findIndex(column => column.key === key);
       },
       getColumnIndexFromColumnKey(columnKey) {
-        return this.columns.findIndex(column => column.columnDataIndex !== null && ResultSetDataKeysUtils.isEqual(columnKey, column.columnDataIndex));
+        return columns.findIndex(column => column.columnDataIndex !== null && ResultSetDataKeysUtils.isEqual(columnKey, column.columnDataIndex));
```

Comment:
this code won't work well because it will capture only first `columns` object value